app/models/messaging/bsnl/dlt_template.rb

```diff
@@ -0,0 +1,108 @@
+class Messaging::Bsnl::DltTemplate
+  MAX_VARIABLE_LENGTH = 30
+  TRIMMABLE_VARIABLES = %i[facility_name patient_name].freeze
+  BSNL_TEMPLATES = YAML.load_file("config/data/bsnl_template_details.yml")
+  STATUSES = {
+    "0" => :pending_naming,
+    "1" => :approved
+  }
+
+  attr_reader :name
+  attr_reader :id
+  attr_reader :keys
+  attr_reader :max_length_permitted
+  attr_reader :non_variable_text_length
+  attr_reader :variable_length_permitted
+  attr_reader :status
+
+  def initialize(dlt_template_name)
+    @name = dlt_template_name
+    details = template_details(dlt_template_name)
+    @id = details["Template_Id"]
+    @status = STATUSES[details["Template_Status"]]
+    @keys = details["Template_Keys"]
+    @max_length_permitted = details["Max_Length_Permitted"].to_i
+    @non_variable_text_length = details["Non_Variable_Text_Length"].to_i
+    @variable_length_permitted = max_length_permitted - non_variable_text_length
+  end
+
+  def massaged_variable_content(content)
+    content.compact
+      .then { |c| check_variables_presence(c) }
+      .then { |c| trim_variables(c) }
+      .then { |c| limit_total_variable_length(c) }
+      .then { |c| check_total_variable_length(c) }
+  end
+
+  def check_variables_presence(content)
+    missing_keys = missing_keys(content)
```

Comment:
Careful naming the local the same as the method 😅 Let's switch it up if possible.

Reply:
Wups, thanks. Fixed in [fa229da](https://github.com/simpledotorg/simple-server/pull/3649/commits/fa229daf052cafd744a2a12cd939a34e108f31c8)